recipes/cgal/all/test_package/conanfile.py

```diff
@@ -1,20 +1,26 @@
 import os
 
 from conan import ConanFile
-from conan.tools import build
-from conans import CMake
-
+from conan.tools.cmake import CMake, cmake_layout
+from conan.tools.build import can_run
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+    generators = "CMakeDeps", "CMakeToolchain"
+
+    def requirements(self):
+        self.requires("eigen/3.4.0")
```

Comment:
Some parts of CGAL require Eigen3, but not CGAL core parts.

Reply:
Is it something new? We didn't have it before and cgal was working. If it's a new dependency, it would require a `if Version(self.version) < "5.5.2"` for instance.

Reply:
It should be solved as cgal dependency directly then, and could be added as dependency based on an option.

Reply:
CGAL is a header-only library, with dependencies and optional dependencies. That is not new. What is new is that I have upgraded the `test_package` with a less trivial test.

During the build of the CGAL Conan package, the dependencies will not change anything in the produced package (that is header-only). I do not think Conan package options are a good way to deal with that. There are only three dependencies: boost, gmp, and mpfr. But currently **18** optional dependencies. That is not new.

If a user has a program with several compilation units using CGAL (like the CGAL 3D demo, with all its plugins) but only some of them using optional dependencies, then the use of Conan option would pull the dependencies for all the compilation units, and not only for the one really needing them.